### PR TITLE
Cleanup transforms

### PR DESCRIPTION
**Pull request by homm**
Internal transform API was very confusing. For example, `rotate` method was calling `self.transform` or `self.im.rotate` depends on `expand` argument, but `self.im.rotate` internally also calls `ImagingTransformAffine`, so it was excessive.

Changes proposed in this pull request:

 * First of all, this PR fixes rotations bug #1917 in two ways:
    * It really uses very cheap `transpose` operations if it is possible regardless of filter. https://github.com/python-pillow/Pillow/pull/1941/commits/8203a43d262d349f71bcfa3da46e5f3e30fe99e0
    * It rounds coefficients in the transform matrix, so even if we are not using cheap transpose, we still get right dimensions because `cos(90°)` now is exactly 0, not 6.123e-17. https://github.com/python-pillow/Pillow/pull/1941/commits/7687ce829e50635838707f5153079fb4d34cd41e
 * `Image.rotate` always uses `self.transform`, so `im.rotate` was removed from core library. https://github.com/python-pillow/Pillow/pull/1941/commits/709078efd14cfcffaedf59e883f56aa16d61926e https://github.com/python-pillow/Pillow/pull/1941/commits/1f8c2527baaff6357f31761b6340d2016920781f
 * The order of coefficients in internal API now match matrix representation and order of external API. https://github.com/python-pillow/Pillow/pull/1941/commits/2b77b1cec7fae15112be03dd5c1bfbb0169c3e0b https://github.com/python-pillow/Pillow/pull/1941/commits/3d622d60cf7c6c609cfc48b774ee0d8aef2d1d85
 * Internal `ImagingTransformAffine`, `ImagingTransformPerspective` and `ImagingTransformQuad` was replaced with one entry point to all transformations: `ImagingTransform`. Former `ImagingTransform` was  renamed to `ImagingGenericTransform` and removed from extern symbols (`Imaging.h` file). https://github.com/python-pillow/Pillow/pull/1941/commits/6be3df2a432bdc61d0705ebacfc878b93cc86b9c https://github.com/python-pillow/Pillow/pull/1941/commits/9902d2e1c50712a7fa1d308fab124bf5c71f018e

### CODE REVIEW

**Comment by homm** [comment]
rebased

File: PIL/Image.py [hunk 1]
```diff
@@ -1863,28 +1865,24 @@ def transform(self, size, method, data=None, resample=NEAREST, fill=1):
...
             # convert extent to an affine transform
             x0, y0, x1, y1 = data
             xs = float(x1 - x0) / w
             ys = float(y1 - y0) / h
             method = AFFINE
-            data = (x0 + xs/2, xs, 0, y0 + ys/2, 0, ys)
+            data = (xs, 0, x0 + xs/2, 0, ys, y0 + ys/2)
+
         elif method == PERSPECTIVE:
             # change argument order to match implementation
```
Comment (wiredfool) [inline PIL/Image.py]: This comment doesn't apply anymore, right?

Comment (homm) [inline PIL/Image.py]: Yes, removed

**Comment by homm** [comment]
Links to commits in the description are updated.

**Pull request merged**